src/backend/joanie/tests/lms_handler/test_backend_moodle.py

```diff
@@ -0,0 +1,296 @@
+        "archetype": "manager",
+    },
+    {
+        "id": 2,
+        "name": "",
+        "shortname": "coursecreator",
+        "description": "",
+        "sortorder": 2,
+        "archetype": "coursecreator",
+    },
+    {
+        "id": 3,
+        "name": "",
+        "shortname": "editingteacher",
+        "description": "",
+        "sortorder": 3,
+        "archetype": "editingteacher",
+    },
+    {
+        "id": 4,
+        "name": "",
+        "shortname": "teacher",
+        "description": "",
+        "sortorder": 4,
+        "archetype": "teacher",
+    },
+    {
+        "id": 5,
+        "name": "",
+        "shortname": "student",
+        "description": "",
+        "sortorder": 5,
+        "archetype": "student",
+    },
+    {
+        "id": 6,
+        "name": "",
+        "shortname": "guest",
+        "description": "",
+        "sortorder": 6,
+        "archetype": "guest",
+    },
+    {
+        "id": 7,
+        "name": "",
+        "shortname": "user",
+        "description": "",
+        "sortorder": 7,
+        "archetype": "user",
+    },
+    {
+        "id": 8,
+        "name": "",
+        "shortname": "frontpage",
+        "description": "",
+        "sortorder": 8,
+        "archetype": "frontpage",
+    },
+]
+
+MOODLE_RESPONSE_USERS = {
+    "users": [
+        {
+            "id": 5,
+            "username": "student",
+            "firstname": "Student",
+            "lastname": "User",
+            "fullname": "Student User",
+            "email": "student@example.com",
+            "department": "",
+            "firstaccess": 1704716076,
+            "lastaccess": 1704716076,
+            "auth": "manual",
+            "suspended": False,
+            "confirmed": True,
+            "lang": "en",
+            "theme": "",
+            "timezone": "99",
+            "mailformat": 1,
+            "description": "",
+            "descriptionformat": 1,
+            "profileimageurlsmall": (
+                "https://moodle.test/theme/image.php/boost/core/1704714971/u/f2"
+            ),
+            "profileimageurl": (
+                "https://moodle.test/theme/image.php/boost/core/1704714971/u/f1"
+            ),
+        }
+    ],
+    "warnings": [],
+}
+
+
+@override_settings(
+    JOANIE_LMS_BACKENDS=[
+        {
+            "API_TOKEN": "a_secure_api_token",
+            "BACKEND": "joanie.lms_handler.backends.moodle.MoodleLMSBackend",
+            "BASE_URL": "http://moodle.test/webservice/rest/server.php",
+            "COURSE_REGEX": r"^.*/course/view.php\?id=.*$",
+            "SELECTOR_REGEX": r"^.*/course/view.php\?id=.*$",
+        },
+        {
+            "API_TOKEN": "a_secure_api_token",
+            "BACKEND": "joanie.lms_handler.backends.openedx.OpenEdXLMSBackend",
+            "BASE_URL": "http://openedx.test",
+            "COURSE_REGEX": r"^.*/courses/(?P<course_id>.*)/course/?$",
+            "SELECTOR_REGEX": r"^.*/courses/(?P<course_id>.*)/info/?$",
+        },
```

Comment:
Is this useful for this file ? I don't see you any calls to the OpenEdX configuration in the following tests.

Reply:
It is just for checking that the right lms is selected, but I added it before seeing test_handler.
We can remove it, I suppose.